### DISCUSSION TITLE
Modify remote.txt and make a Custom remote? Watch different SB?

### DISCUSSION THREAD

**Discussion by Alimond52** [GerryDazoo/Slinger#556] [Q&A]
Things are moving forward here. Now all my 4 SB are working. I have started to get the remotes working as well.
Some questions:
1) I have all my IR codes for my STB. I have tried replacing a function on **remote.txt** with a new code and it works. I use Slinger Player 4, which works well and it is displayed on a HD screen 1080 from a PC. When I click on the remote control, the window appears and the buttons are visible. But all are enormously **large**. Even if I expand the Remote window all the way down, all the buttons are not visible. How to make a window that show all **buttons** that window are not higher than **1000 points**? Is it possible to **rearrange** the buttons, so that the buttons you use the most are at the top? How-to? I can edit a txt file but am not that good at HTML.
2) The **size** of the video window and the remote control need to be matched a little more. Suggests a slightly **larger** **video** window and together with the remote control window covers the **width** of the monitor size **1920**.
3) Slinger Player 4 would need a **back** command from full screen. From what I can see, there is only "**Esc**" at the moment. If you are sitting on a TV sofa, you can easily have a physical **mouse** available. But maybe not a **keyboard** on your lap? On the old SlingPlayer 2.04, it was easy to just **double-click** with the mouse on full screen and it went down in size and vice versa on the next double-click. I don't know how difficult it is to add such a function?
4) Those of you who made your **own** natural like remote control that resembles the physical one. How have you **done**? I have a Humax remote called **RM-301**, and it worked great for ProHD and SB500 since 2011. I have saved most of those files from the Slingplayer days. Can one **use** those files in any way? Or do you have to **redo** everything from **scratch** to fit for the Slinger Player?
5) I have **4 different SB**s with different names that I want to see in Slinger Player. What is the best way to **quickly** **switch** between each SB?
Thanks for any help.

**Comment by watanyoshi** [comment]
For 3, double-click feature is already in the latest Windows player and it works very well with AoT option.
For Remote customization, I had the same problem with big buttons but was able to manage by editing the remote.txt. I will share it for your reference here.
[remote_temp.txt](https://github.com/GerryDazoo/Slinger/files/10935698/remote_temp.txt)

**Reply by Alimond52** [reply]
Thanks so much!!!! Is it also possible to change the order of the buttons displayed in the window? So more frequent used buttons are on the top?

**Reply by watanyoshi** [reply]
Of course you can change the order of buttons. My remote looks like this. You can change the color and use Emoji if you want.

![myRemote](https://user-images.githubusercontent.com/51803484/224204624-e14ad1c4-23eb-4780-8ce0-8454d4182f7e.png)

**Reply by Alimond52** [reply]
Wow, it looks great!!!
I was just going to ask how to change to symbols on the buttons instead of letters.
But you had already fixed it.
I want to use your remote. I'm traveling abroad in 3 days so I probably won't have time to fix it before then.
You have some buttons I'm a little unsure about, my old Humax STB probably doesn't have those functions?
Such as:
**Skip>>**, isn't that where you skip forward a few minutes?
**C ^**, are there channels up?
**P ^**, what is it?
Thank you so much for sharing!!!!

**Reply by watanyoshi** [reply]
Hey Alimond52, I am glad to help more. If you have old log files of Sling Player, you will find the command lines you used in the files. They look like:
triggerRemoteKeyPressed event - id.remote.power key - 1
triggerRemoteKeyPressed event - id.remote.guide key - 35
For the keys you asked...
Skip>>:  triggerRemoteKeyPressed event - id.remote.skip key - 30
C ^: triggerRemoteKeyPressed event - id.remote.channel.up key - 4
P ^: triggerRemoteKeyPressed event - id.remote.page.up key - 43
The log files should contain almost all the keys you use in your regular operations. Scan the log files to find keys you need on your custome remote.

Also, you should be able to find compete key mapping in a file:
C:\Users\<..your name..>\AppData\Roaming\Sling Media\WebSlingPlayer\Remote\Default.CABLEBOX.json

I have used generic remote [Remote7](http://www.hifi-remote.com/wiki/index.php/Customising_the_Remote_Image_on_your_Slingplayer) for a long time instead of the actual remote for my STB then Page Up/Down actual operations are different from the labels. It is all up to you how you order the buttons and what you label and what you assign the code to each button. Hope you can enjoy the work as I did.

**Comment by mike-uk** [comment]
How do you set up custom remote for unified_config.ini setup? I have put the custom remote txt in /home/slingbox/

The standard setup with just one sb is like below and it works
`code=0`
`include=/home/slingbox/ProHD_Tuner.txt`

for more than one box chang to remote= so I changed to as below but doesn't work. It only work when using remote.txt where is the remote.txt located on raspberry pi install? I think it is in part of the main code instead of an actual txt file.
`Remote=/home/slingbox/ProHD_Tuner.txt`

**Reply by highfivin** [reply]
I put the remotes in the same directory as my slinger server and config.ini
Then I just do this
Remote=ProHD_Tuner.txt

**Comment by highfivin** [comment]
Here is the remote I modified to work with my HR24 DVR

![Screenshot 2023-05-17 at 6 48 12 PM](https://github.com/GerryDazoo/Slinger/assets/133918667/bcb5ce57-9be6-48d2-bb4d-a91a591c344b)

**Comment by as400jockey** [comment]
Slingerplayer has been awesome with my Verizon Fios.  I'm running server 4.01 on Windows.  Accessing it using the Android app on a tablet, and also on a firestick.

Now, Fios is going away.

I am trying to connect an over-the-air tuner box to my Slingbox.  It's all connected, and now I only need to cobble together a remote control.

The tuner box is called ZJBox.  I found what I think are the remote codes here:

https://dev.to/raleighlittles/zjbox-tv-converter-remote-communication-guide-ha7

Can someone point me in the direction of how to cobble this together?

**Comment by as400jockey** [comment]
or  would a gadget like this help to get the codes:

https://www.amazon.com/Generic-Tv-Remote-Control-Decoder/dp/B00PP6GRPG/ref=sr_1_3?crid=X6NEVX2TVQEU&keywords=remote+control+decoder&qid=1702585365&sprefix=remote+control+decoder%2Caps%2C139&sr=8-3

**Comment by as400jockey** [comment]
please continue this discussion here:

https://github.com/GerryDazoo/Slinger/discussions/641
